presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -88,12 +130,56 @@ public String explain(CreateTable statement)
                 session,
                 metadata);
 
+        combineProperties(finalProperties, statement.getProperties().keySet(), properties);
+
         TableMetadata tableMetadata = new TableMetadata(
                 tableName.getCatalogName(),
-                new ConnectorTableMetadata(tableName.asSchemaTableName(), columns, properties, session.getUser(), false));
+                new ConnectorTableMetadata(tableName.asSchemaTableName(), columns, finalProperties, session.getUser(), false));
 
         metadata.createTable(session, tableName.getCatalogName(), tableMetadata);
 
         return completedFuture(null);
     }
+
+    private void mergeTableProperties(Map<String, Object> source, Map<String, Object> destination, Optional<List<LikeClause.LikeOption>> likeOptions)
+    {
+        if (likeOptions.isPresent()) {
+            List<LikeClause.LikeOption> options = likeOptions.get();
+            Set<LikeClause.LikeOption.Property> optionProperties = new HashSet<>();
+            boolean includingAll = false;
+            for (LikeClause.LikeOption option : options) {
+                LikeClause.LikeOption.Property likeProperty = option.getProperty();
+                if (optionProperties.contains(likeProperty)) {
+                    throw new PrestoException(StandardErrorCode.SYNTAX_ERROR, "the same LIKE option should only be specified once: " + option.getType());
+                }
+                optionProperties.add(likeProperty);
+
+                if (option.getType().equals(LikeClause.LikeOption.Type.INCLUDING)) {
+                    if (likeProperty.equals(ALL)) {
+                        includingAll = true;
+                        break;
+                    }
+                    else if (LIKE_INCLUDING_PROPERTIES.containsKey(likeProperty)) {
```

Comment:
For `EXCLUDING`, I think we should just ignore this property and do nothing to it since `EXCLUDING` means that we don't want to include this property in the new created table.